### DISCUSSION TITLE
Is 'keycloak-js' deprecated as well

### DISCUSSION THREAD

**Discussion by StefanKern** [keycloak/keycloak#11975] [Adapter deprecation]
With the [adapter deprecation] (https://www.keycloak.org/2022/02/adapter-deprecation.html ) we where wondering if ['keycloak-js'](https://www.npmjs.com/package/keycloak-js ) is getting deprecated as well?

In the article it is not mentioned by name, and we are not sure if it counts as an 'adapter'. Also in the npm package, there is no note yet of it's deprecation, but in the link to the documentation where it is called and adapter, but also no deprecation hint there.

So currently I'm only 65% sure that it will get depracted.

And if so, it's replacement most likely is [openid-client](https://www.npmjs.com/package/openid-client).

Thank you

**Comment by stianst** [comment]
No, it's not being deprecated

**Comment by stianst** [comment]
Please see https://www.keycloak.org/2023/03/adapter-deprecation-update.html for an updated plan for adapter deprecation. In summary for the client-side JavaScript adapter that remains. We do plan to look for alternatives, or to revamp the adapter at some point in the future though.

**Comment by Zysberg** [comment]
Are there any updates to this? it's been well over a year.

**Reply by jonkoops** [reply]
No, there are no updates because as mentioned we're not deprecating this package.

**Comment by mariomurrent-softwaresolutions** [comment]
If the keycloak-js won't be deprecated maybe there is a chance that cordova & cordova-native will be upgraded to capacitor

**Reply by jonkoops** [reply]
We're actually more likely to deprecate and remove the Cordova and Cordova Native adapters. This is because we have no testing infrastructure for these adapters, nor any maintainers that are interested in maintaining them.

Keycloak JS allows users to write [custom adapters](https://www.keycloak.org/securing-apps/javascript-adapter#custom-adapters) to integrate with Keycloak JS, so if someone from the community wants to write a Capacitor adapter that is entirely possible.

**Reply by mariomurrent-softwaresolutions** [reply]
Ah I see, so you completely remove it from the code base?

Well for writing some custom adapters for Capacitor, and so on, some more methods need to be exposed.

So the go to would be, write an adapter not put it into the keycloak-js library?

**Reply by jonkoops** [reply]
Yes, essentially we'd prefer those are maintained by the community as custom adapters that are seperate, but can be plugged into Keycloak JS. I'd like to generalize the adapters even more so they could also be more capable than they are now, but this will all depend on our bandwidth and community feedback.

**Reply by mariomurrent-softwaresolutions** [reply]
Sounds like a good plan. So if I need two more methods exposed I can probably open a PR for that?

**Reply by jonkoops** [reply]
Yes, feel free to propose further extensions in the form of a PR. Do note that at the moment we're in a freeze for Keycloak JS, as we moved to a [new repository](https://github.com/keycloak/keycloak-js) and we're in the process of porting over the old Java based test-suite. Once the test suite is in place we'll be accepting new work again.
